File: scripts/_resolve-skill-script.js
/**
 * Resolve a skill script path across cloned-repo and installed CLI layouts.
 */
const fs = require('fs');
const path = require('path');

function resolveSkillScript(skillName, scriptName) {
  const roots = [
    path.join(__dirname, '..', 'skills', skillName, 'scripts', scriptName),
    path.join(__dirname, '..', '.agents', 'skills', skillName, 'scripts', scriptName),
    path.join(__dirname, '..', '.grok', 'skills', skillName, 'scripts', scriptName),
    path.join(__dirname, '..', '.claude', 'skills', skillName, 'scripts', scriptName),
    path.join(__dirname, '..', '.codex', 'skills', skillName, 'scripts', scriptName),
    path.join(__dirname, '..', '.skills', skillName, 'scripts', scriptName),
  ];

  for (const candidate of roots) {
    if (fs.existsSync(candidate)) return candidate;
  }
  return null;
}

function runSkillScript(skillName, scriptName) {
  const target = resolveSkillScript(skillName, scriptName);
  if (!target) {
    console.error(`❌ Could not find ${skillName}/scripts/${scriptName}. Run install-skills.js first.`);
    process.exit(1);
  }

  const { spawnSync } = require('child_process');
  const result = spawnSync(process.execPath, [target, ...process.argv.slice(2)], {
    stdio: 'inherit',
  });
  process.exit(result.status ?? 1);
}

module.exports = { resolveSkillScript, runSkillScript };

File: scripts/_resolve_skill_script.py
#!/usr/bin/env python3
"""Resolve skill script paths across cloned-repo and installed CLI layouts."""
import os
import subprocess
import sys

INSTALL_ROOTS = (".agents", ".grok", ".claude", ".codex", ".skills")


def resolve_skill_script(skill_name, script_name, root=None):
    if root is None:
        root = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")

    candidates = [os.path.join(root, "skills", skill_name, "scripts", script_name)]
    for install_root in INSTALL_ROOTS:
        candidates.append(
            os.path.join(root, install_root, "skills", skill_name, "scripts", script_name)
        )

    return next((p for p in candidates if os.path.exists(p)), None)


def run_skill_script(skill_name, script_name):
    target = resolve_skill_script(skill_name, script_name)
    if not target:
        print(
            f"❌ Could not find {skill_name}/scripts/{script_name}. Run install-skills.js first.",
            file=sys.stderr,
        )
        sys.exit(1)

    result = subprocess.run([sys.executable, target, *sys.argv[1:]])
    sys.exit(result.returncode)


if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage: _resolve_skill_script.py <skill-name> <script-name> [args...]", file=sys.stderr)
        sys.exit(1)
    run_skill_script(sys.argv[1], sys.argv[2])